pkg/helm/http_renderer.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+
+	"istio.io/pkg/log"
+)
+
+// HTTPTemplateRenderer is a helm template renderer for helm charts hosted on http url
+type HTTPTemplateRenderer struct {
+	namespace        string
+	componentName    string
+	helmChartDirPath string
+	chart            *chart.Chart
+	started          bool
+	globalValues     string
+	pollInterval     time.Duration
+}
+
+// NewHTTPTemplateRenderer creates a TemplateRenderer with the given parameters and returns a pointer to it.
+// helmChartDirPath must be an absolute HTTP URL hosting root of helm charts.
+func NewHTTPTemplateRenderer(helmChartDirPath, globalValues, componentName, namespace string, pollInterval time.Duration) *HTTPTemplateRenderer {
```

Comment:
we also need a non-polling mode. somewhere in the comments you should describe what the HTTPTemplateRenderer actually does.